File: app/src/main/java/com/qns/data/local/dao/ChatDao.java
package com.qns.data.local.dao;
import androidx.room.*;
import com.qns.data.local.entity.ChatEntity;
import java.util.List;
import io.reactivex.rxjava3.core.*;
@Dao
public interface ChatDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE) Completable upsert(ChatEntity c);
    @Insert(onConflict = OnConflictStrategy.REPLACE) Completable upsertAll(List<ChatEntity> cs);
    @Query("SELECT * FROM chats ORDER BY lastMessageAt DESC") Flowable<List<ChatEntity>> getAll();
    @Query("SELECT * FROM chats WHERE id = :id LIMIT 1") Flowable<ChatEntity> getById(String id);
    @Query("UPDATE chats SET otherUserOnline = :online WHERE otherUserId = :uid") Completable setOnline(String uid, boolean online);
    @Query("UPDATE chats SET unreadCount = unreadCount + 1 WHERE id = :id") Completable incUnread(String id);
    @Query("UPDATE chats SET unreadCount = 0 WHERE id = :id") Completable clearUnread(String id);
    @Query("DELETE FROM chats WHERE id = :id") Completable delete(String id);
}

File: qunes-android-app/src/main/kotlin/com/qunes/app/data/local/dao/ChatDao.kt
package com.qunes.app.data.local.dao

import androidx.room.*
import com.qunes.app.data.local.entities.ChatEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface ChatDao {
    @Query("SELECT * FROM chats ORDER BY updatedAt DESC")
    fun getAllChats(): Flow<List<ChatEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertChat(chat: ChatEntity)

    @Delete
    suspend fun deleteChat(chat: ChatEntity)
}